accounts/abi/bind/auth.go

```diff
@@ -45,14 +56,23 @@ func NewTransactor(keyin io.Reader, passphrase string) (*TransactOpts, error) {
 }
 
 // NewKeyStoreTransactor is a utility method to easily create a transaction signer from
-// a decrypted key from a keystore.
+// an decrypted key from a keystore.
+//
+// Deprecated: Use NewKeyStoreTransactorWithChainID instead.
 func NewKeyStoreTransactor(keystore *keystore.KeyStore, account accounts.Account) (*TransactOpts, error) {
+	log.Warn("WARNING: NewKeyStoreTransactor has been deprecated in favour of NewTransactorWithChainID")
+	var signer types.Signer = types.HomesteadSigner{}
 	return &TransactOpts{
 		From: account.Address,
-		Signer: func(signer types.Signer, address common.Address, tx *types.Transaction) (*types.Transaction, error) {
+		Signer: func(address common.Address, tx *types.Transaction) (*types.Transaction, error) {
 			if address != account.Address {
-				return nil, errors.New("not authorized to sign this account")
+				return nil, ErrNotAuthorized
+			}
+			// Quorum
+			if tx.IsPrivate() {
+				signer = types.QuorumPrivateTxSigner{}
 			}
+			// / Quorum
```

Comment:
```suggestion
			// Quorum
			signer := Signer
			if tx.IsPrivate() {
				signer = types.QuorumPrivateTxSigner{}
			}
			// / Quorum
```
need to copy the value